node/src/manager/deployment.rs

```diff
@@ -179,11 +181,17 @@ pub fn find_single_deployment_locator(
     pool: &ConnectionPool,
     name: &str,
 ) -> anyhow::Result<DeploymentLocator> {
-    let deployment_locator = match &Deployment::lookup(pool, name)?[..] {
-        [] => anyhow::bail!("Found no deployment for the given ID"),
-        [deployment_locator] => deployment_locator,
-        _ => anyhow::bail!("Found multiplle deployments for given identifier"),
-    }
-    .locator();
+    let mut locators: Vec<DeploymentLocator> = HashSet::<DeploymentLocator>::from_iter(
+        Deployment::lookup(pool, name)?
+            .into_iter()
+            .map(|deployment| deployment.locator()),
+    )
+    .into_iter()
+    .collect();
+    let deployment_locator = match locators.len() {
+        0 => anyhow::bail!("Found no deployment for `{}`", name),
+        1 => locators.pop().unwrap(),
+        n => anyhow::bail!("Found {} deployments for `{}`", n, name),
+    };
```

Comment:
Interesting!
I didn't realize a deployment could have multiple names.

Reply:
Yeah, that's what keeps things interesting. When people deploy an IPFS hash that was already running under a different name, we just put an entry in `subgraph` and `subgraph_version` to point to that deployment. So the subgraph name -> deployment mapping is many-to-one